types/src/transaction/transaction_v1/transaction_v1_builder.rs

```diff
@@ -327,6 +361,18 @@ impl<'a> TransactionV1Builder<'a> {
         }
     }
 
+    /// Returns a reference to the `body` field of the transaction.
+    ///
+    /// This method provides access to the transaction's body, which contains
+    /// the core data of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// A reference to the `TransactionV1Body`.
+    pub fn body(&self) -> &TransactionV1Body {
```

Comment:
what's the reasoning behind exposing the incomplete transaction body from the builder?

Reply:
I believe it was inaccessible from outside and I could not read the existent to modify a transaction like 

```rust
    pub fn with_entity_hash(
        &self,
        hash: AddressableEntityHash,
        secret_key: Option<String>,
    ) -> Transaction {
        self.build(BuildParams {
            secret_key,
            body: Some(modify_body(
                &self.body(),
                NewBodyParams {
                    new_hash: Some(hash),
                    ..Default::default()
                },
            )),
            ..Default::default()
        })
    } 
```
with self body returning

`transaction_v1.body()`

Reply:
to me it seems like we should not depend on data from inside the builder to use the builder, the builder's methods should allow you to modify its contents

Reply:
in general exposing the contents of a builder is somewhat of an antipattern

Reply:
Well I don't think there is a need inside for node and client to expose/manipulate this body as there is a need outside of them so I believe this was the shortest getter to have what constitutes it chain_name, timestamp, ttl, body,pricing_mode, initiator_addr) so why not body ? without too much over complication. I believe the body is also constituted from many things so it would imply a setter and a getter for each to modify the body., so basically port the sdk code into the client while the client not using it.

Reply:
The builder is exposing everything expect the body so the anti pattern is non only on the body as it is a nested sytructure.

Reply:
the builder isn't currently exposing any of its fields, it's meant to be populated with the `with_*` methods and then built with the `build` method, you can pass any additional information along instead of accessing it from inside the builder

Reply:
Yes thanks so do do you want to have additional methods covring the body fields the same way ? the client does not need this.